### PR TITLE
Feature/export generic input shape

### PR DESCRIPTION
**Pull request by lorenzomammana**
## Summary

Integrate automatic computation of input shape using an nn.Module decorator, refactor input_size export for deployment model json, refactor mlflow export to adapt with the input size changes.

(There is also a change for ModelManager description that I mistakenly put in this branch 😶)

## Type of Change

Please select the one relevant option below:

- Breaking change

## Checklist

Please confirm that the following tasks have been completed:

- [X] I have tested my changes locally and they work as expected.
- [x] I have added unit tests for my changes, or updated existing tests if necessary.
- [x] I have updated the documentation, if applicable.
- [X] I have installed pre-commit and run locally for my code changes.

## Additional Information (Optional)

Now input size is a List of Dictionaries, Tuples or other Lists, the base element will be tuples representing the input shape withouth batch size.

E.g.

Input_size = [(3, 224, 224), [(3, 224, 224), (3, 224, 224)], {"x": (3, 224, 224)}]

### CODE REVIEW

File: quadra/models/base.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import inspect
+from typing import Any, Sequence
+
+import torch
+from torch import nn
+
+
+class ModelWrapper(nn.Module):
```
Comment (AlessandroPolidori) [inline quadra/models/base.py:10]: In my opinion ModelWrapper is too general. You could call it "Foo" at this point (revenge is a dish best served cold)

Comment (lorenzomammana) [reply quadra/models/base.py:10]: I agree on this, what do you suggest as a different name?

File: quadra/tasks/segmentation.py [hunk 2]
```diff
@@ -113,8 +115,8 @@ def prepare(self) -> None:
     def export(self) -> None:
         """Generate a deployment model for the task."""
         log.info("Exporting model ready for deployment")
-        input_width = self.config.transforms.get("input_width")
-        input_height = self.config.transforms.get("input_height")
+        self.config.transforms.get("input_width")
```
Comment (AlessandroPolidori) [inline quadra/tasks/segmentation.py:118]: I don't understand where are they stored now

File: quadra/tasks/segmentation.py [hunk 3]
```diff
@@ -136,19 +138,38 @@ def export(self) -> None:
...
             log.info("idx_to_class is present")
             classes = self.config.datamodule.idx_to_class
+
         if self.export_type is None:
             raise ValueError(
                 "No export type specified. This should not happen, please check if you have set "
                 "the export_type or assign it to a default value."
             )
+
+        # TODO: Take it from the config
```
[the remark this answers is not in the document]
Comment (lorenzomammana) [reply quadra/tasks/segmentation.py:148]: Two reasons mainly, there may be cases on which the forward due to a weird implementation (like if elses in train/val) doesn't give the right output to the wrapping model, second maybe the user wants to have some custom shape for export for unknown reasons.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -1,13 +1,13 @@
 fail_fast: true
 
 repos:
-  - repo: local
```
Comment (lorenzomammana) [inline .pre-commit-config.yaml:4]: I had to disable ruff due to an issue with yaml files which I can't understand :|

Comment (lorenzomammana) [reply .pre-commit-config.yaml]: In the end I put ruff working only for .py files, I checked on master and the error on yaml is there :/

**Comment by lorenzomammana** [comment]
Turned out to be quite a huge PR, a lot of recursion also 😮
I've wrote many tests so I hope it's will hold fine for a while.

**Pull request merged**